modules/authentication/proc-creating-a-custom-transformer-to-provision-users-from-rhsso-to-the-software-catalog.adoc

```diff
@@ -0,0 +1,92 @@
+[id="provisioning-users-from-rhsso-to-the-software-catalog"]
+= Creating a custom transformer to provision users from Red Hat Single-Sign On (RHSSO) to the software catalog
+
+To customize further how users/groups are mapped into entities in {product}, you can create a custom backend module that utilizes the `keycloakTransformerExtensionPoint` to provide custom user and group transformers to the keycloak backend.
+
+.Prerequisites
+* You xref:provisioning-users-from-rhsso-to-the-software-catalog[enabled provisioning users from Red Hat Single-Sign On (RHSSO) to the software catalog].
+
+.Procedure
+. Create a new backend module with the `yarn new` command.
+
+. Add your custom user and group transformers to the `keycloakTransformerExtensionPoint`.
+
++
+Below is an example of how the backend module can be defined:
++
+.`plugins/__<module-name>__/src/module.ts`
+[source,javascript]
+----
+/* highlight-add-start */
+import {
+  GroupTransformer,
+  keycloakTransformerExtensionPoint,
+  UserTransformer,
+} from '@janus-idp/backstage-plugin-keycloak-backend';
+
+const customGroupTransformer: GroupTransformer = async (
+  entity,
+  realm,
+  groups,
+) => {
+  /* apply transformations */
+  return entity;
+};
+const customUserTransformer: UserTransformer = async (
```

Comment:
Consider adding comments that explain each arg of the transformer ([see type def](https://github.com/janus-idp/backstage-plugins/blob/ffb600d576c35f207f530817ba11a42a68066661/plugins/keycloak-backend/src/lib/types.ts#L37-L54))

Reply:
@JessicaJHee Like this? (I am unsure of the correct syntax):

```suggestion
/**
 * Customize the ingested User entity
 *
 * entity: The output of the default parser
 * user: Keycloak user representation
 * realm: Realm name
 * groups: Data about available groups (can be used to create additional relationships)
 */
const customUserTransformer: UserTransformer = async (
```

Reply:
const customUserTransformer: UserTransformer = async (
  entity, // entity output from default parser
  user, // Keycloak user representation
  realm, // Keycloak realm name
  groups, // Keycloak group representation
) => {
  /* apply transformations */
  return entity;
};
/* highlight-add-end */